### ISSUE TITLE
Non finalized zarr in 000026 but dandiset report does not complain about assets

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-archive#2121]
- #2091

was fixed recently and from my reading we should have https://dandiarchive.org/dandiset/000026?search=000026&pos=2 reporting an error since one of its zarrs is still in Pending state and was never finalized (details are at https://github.com/dandi/dandi-cli/issues/1553#issuecomment-2556234954).

Should revalidation of all dandisets be triggered to may be update the state?  WDYT @jjnesbitt ?

edit: specifically this zarr is not complete yet:

```shell
$ curl --silent -X 'GET'   'https://api.dandiarchive.org/api/zarr/7933bc72-ec3f-4043-8076-d7aa8a70e4f7/'   -H 'accept: application/json' | jq .
{
  "name": "sub-I48/ses-SPIM/micr/sub-I48_ses-SPIM_sample-BrocaAreaS03_stain-Somatostatin_SPIM.ome.zarr",
  "dandiset": "000026",
  "zarr_id": "7933bc72-ec3f-4043-8076-d7aa8a70e4f7",
  "status": "Pending",
  "checksum": null,
  "file_count": 0,
  "size": 0
}
```

**Comment by yarikoptic** [comment]
FTR 03032025: I have fixed one contributor's name (validator was complaining) and now dandiset should get revalidated

**Comment by jjnesbitt** [comment]
The issue you linked (#2091) does not specify that non-finalized zarrs should show up as a validation error in the dandiset. What is says is that _if_ you finalize a zarr, it should set the draft version back to `PENDING` so it can be re-validated. 

What you're asking for (and what it seems is being stated in https://github.com/dandi/dandi-archive/issues/2129) is that any zarrs that have a status of `PENDING` should show up as a validation error on the dandiset itself.

Would you like me to take this on this issue?

**Comment by yarikoptic** [comment]
yes, I might have not realized back when filing this #2121 that we still "permit" having non-finalized zarrs, and realization came later in #2129.
Even though we still can't publish dandisets with zarrs, I think it is pre-requisite to be able to catch those non-finalized uploads and inform users about them hence we better have it addressed.
I think that 
- #2129 

is at large a "generalization" of this one, so I will close this in its favor. I would leave it to work out priority in addressing it if you are working on something else atm for you to decide/discuss with @kabilar .
